Fixes #5880 - Move test-simple-webapp to demos.

sbordet opened the pull request:
Moved test-simple-webapp to demos.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet committed:
Fixes #5880 - Move test-simple-webapp to demos.

Moved test-simple-webapp to demos.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

joakime reviewed (approved):
I'm good with it as is.
But it would be better if index.jsp was just renamed to index.html (as you don't have a JSP dependency declared in your demo-simple.mod

sbordet committed:
Renamed index.jsp to index.html since it's a simple HTML page.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet committed:
Issue #5880 - Move test-simple-webapp to demos.

Added demo-simple-jsp-webapp to demos.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

janbartel commented:
@sbordet  if this is moved into demos, then as everything else in demos is deployed by the demos.mod file, then shouldn't this be too?  And shouldn't it be linked onto the root/index.html page as well?

janbartel reviewed (changes requested):
See my comment on the PR.

sbordet commented:
@janbartel I don't think it's worthwhile to add these 2 additional modules to the main demo, as we already have a "hello world" and there is no point in having 3 of them.

They can be enabled individually and they are mostly useful only for the documentation and for testing, so I think it's fine as is.

janbartel commented:
@sbordet well, the hello worlds are actually different: one is a servlet and one is a static page. Also, your simplest webapp isn't the simplest as we don't require you to have a web.xml  if you just have static content :)

sbordet committed:
Issue #5880 - Move test-simple-webapp to demos.

Increased waits for CI runs.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

joakime reviewed (approved):
LGTM

sbordet committed:
Issue #5880 - Move test-simple-webapp to demos.

Prepare to import all the JSP demos.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet committed:
Issue #5880 - Move test-simple-webapp to demos.

Moved JSP demos to their own demo module.
Linked the new module from the main demo and from jetty-home.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

janbartel commented on tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:
```diff
@@ -108,10 +108,10 @@ public void testQuickStartGenerationAndRun() throws Exception
             assertEquals(0, run1.getExitValue());
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
Should this be demo-simple-webapp or the demo-jsp-webapp? There's no demo-simple-jsp-webapp AFAIK.

janbartel commented on tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:
```diff
@@ -159,13 +159,13 @@ public void testSimpleWebAppWithJSP() throws Exception
             assertTrue(Files.exists(jettyBase.resolve("start.ini")));
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
Same as previous comment - think this should be either demo-jsp-webapp or demo-simple-webapp, not sure which.

janbartel commented on tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:
```diff
@@ -198,10 +198,10 @@ public void testSimpleWebAppWithJSPOnModulePath() throws Exception
             assertEquals(0, run1.getExitValue());
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
See previous

janbartel commented on tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:
```diff
@@ -249,10 +249,10 @@ private void testSimpleWebAppWithJSPOverHTTP2(boolean ssl) throws Exception
             assertEquals(0, run1.getExitValue());
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
See previous

janbartel commented on tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:
```diff
@@ -308,7 +308,7 @@ public void testUnixSocket() throws Exception
             assertEquals(0, run1.getExitValue());
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
See previous.

janbartel commented on tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:
```diff
@@ -345,11 +345,11 @@ public void testLog4j2ModuleWithSimpleWebAppWithJSP() throws Exception
             assertTrue(Files.exists(jettyBase.resolve("resources/log4j2.xml")));
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
See previous.

janbartel reviewed (changes requested):
The reference to the webapp is incorrect in several places in the DistributionTests - looks like a simple global search and replace error.

sbordet committed:
Issue #5880 - Move test-simple-webapp to demos.

Fixed reference to demo-jsp from DistributionTests.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

janbartel reviewed (approved):
LGTM

sbordet committed:
Issue #5880 - Move test-simple-webapp to demos.

Fixed reference to demo-jsp from various tests.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>
